### ISSUE TITLE
HTML Attributes vs Component Arguments

### ISSUE THREAD

**Issue by MichalBryxi** [ember-learn/guides-source#1286]
I’m somehow confused about the Component Arguments vs HTML Attributes thing in [built-in components documentation](https://guides.emberjs.com/release/components/built-in-components/)

The `<Input>` component in the examples uses Component Arguments for `@id`, `@type`, `@value`  and HTML Attributes for `disabled`. But then in the table of supported `<input>` attributes one can see `value` and `disabled` next to each other. So it’s not obvious whether one should use `@` symbol or not.

My blind guess would be that the syntax should be:

```hbs
<Input width={{27}} />
<Input placeholder="foo bar" />
<Input maxlength={{42}} />
```

But again seeing the `@type` and `@value` examples hints me that a `@` symbol might be expected for `width`, `placeholder` and `maxlength`?

Should the table of arguments maybe have the respective `@` symbols where Component Argument is expected?

**Comment by locks** [comment]
The built-in components are a bit special because they were designed and implemented much prior to angle bracket syntax and the possibility to pass HTML attributes ad-hoc.
There's also something special regarding `@type` because if I'm remembering correctly, you cannot change the type of an input after adding it to the DOM. There's also some shenanigans going on with the ordering of the attributes. HTML 👀.

We are revisiting this as a whole and likely move it towards favouring HTML attributes, making it more clear which things have to be passed as arguments (e.g. `@type`) and which can be passed as HTML attributes.

**Comment by locks** [comment]
Can you send a PR with the change you mentioned? And tweaking the "You can pass the following standard <input> attributes within the input helper:" prose to make sure it mentions component arguments?

**Comment by jamescdavis** [comment]
Is this closed by #1296?

**Comment by locks** [comment]
Further work is tracked at #1254.
